osquery/events/linux/syslog.cpp

```diff
@@ -0,0 +1,166 @@
+    VLOG(1) << "Pipe does not exist. Creating pipe " << FLAGS_syslog_pipe_path;
+    Status s = createFifo(FLAGS_syslog_pipe_path);
+    if (!s.ok()) {
+      LOG(WARNING)
+          << "Problems were encountered creating the pipe. Please ensure that "
+          << FLAGS_syslog_pipe_path
+          << " exists, is a FIFO, is read/writeable by rsyslog, and is "
+             "readable by this process: "
+          << s.getMessage();
+    }
+  }
+
+  fs::file_status file_status = fs::status(FLAGS_syslog_pipe_path);
+  if (file_status.type() != fs::fifo_file) {
+    return Status(1, "Not a FIFO file: " + FLAGS_syslog_pipe_path);
+  }
+
+  // Opening with both flags appears to be the only way to open the FIFO
+  // without blocking for a writer. We won't ever write to the pipe, but we
+  // don't want to block here and will instead block waiting for a read in the
+  // run() method
+  readStream_.open(FLAGS_syslog_pipe_path,
+                   std::ifstream::in | std::ifstream::out);
+  if (!readStream_.good()) {
+    return Status(1,
+                  "Error opening pipe for reading: " + FLAGS_syslog_pipe_path);
+  }
+  VLOG(1) << "Successfully opened FIFO for syslog ingestion";
+
+  return Status(0, "OK");
+}
+
+Status SyslogEventPublisher::createFifo(const std::string& path) {
+  if (mkfifo(FLAGS_syslog_pipe_path.c_str(), kFifoMode) != 0) {
+    return Status(1, "Error in mkfifo: " + std::string(strerror(errno)));
+  }
+
+  // Explicitly set the permissions since the umask will effect the
+  // permissions created by mkfifo
+  if (chmod(FLAGS_syslog_pipe_path.c_str(), kFifoMode) != 0) {
+    return Status(1, "Error in chmod: " + std::string(strerror(errno)));
+  }
+
+  // Try to set the group so that rsyslog will be able to write to the pipe
+  struct group* group = getgrnam(kFifoGroupName.c_str());
+  if (group == nullptr) {
+    VLOG(1) << "No group " << kFifoGroupName
+            << " found. Not changing group for the FIFO.";
+    return Status(0, "OK");
+  }
+  if (chown(FLAGS_syslog_pipe_path.c_str(), -1, group->gr_gid) == -1) {
+    return Status(1, "Error in chown to group " + kFifoGroupName + ": " +
+                         std::string(strerror(errno)));
+  }
+  return Status(0, "OK");
+}
+
+void SyslogEventPublisher::configure() {
+}
+
+void SyslogEventPublisher::tearDown() {
+}
+
+Status SyslogEventPublisher::run() {
+  while (true) {
+    std::string line;
+    std::getline(readStream_, line);
+    auto ec = createEventContext();
+    Status status = populateEventContext(line, ec);
+    if (status.ok()) {
+      fire(ec);
+    } else {
+      LOG(ERROR) << status.getMessage() << " in line: " << line;
+    }
+  }
+  return Status(0, "OK");
+}
+
+Status SyslogEventPublisher::populateEventContext(const std::string& line,
+                                                  SyslogEventContextRef& ec) {
+  boost::tokenizer<RsyslogCsvSeparator> tokenizer(line);
+  auto key = kCsvFields.begin();
+  for (std::string value : tokenizer) {
```

Comment:
No worries, this works then (for now)!

It would be nice to do some analysis/benchmarking here to find out how many log lines / per second this can handle. But that sort of optimization should not be a blocker.